expression/builtin_cast_vec.go

```diff
@@ -1240,11 +1240,42 @@ func (b *builtinCastJSONAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result
 }
 
 func (b *builtinCastStringAsRealSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastStringAsRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
```

Comment:
It seems that we should check if the args[0] is binary literal [at the begin of this function](https://github.com/pingcap/tidb/blob/master/expression/builtin_cast.go#L1132).

Reply:
PTAL, Thanks. @qw4990 

Checking of binary literal was added. But this checking would be skipped by [here](https://github.com/pingcap/tidb/blob/9a0ac9f7ab8eaf7631e1308e36dbaa532c510922/expression/builtin_cast.go#L168).

I added the checking codes, and left a comment, to make sure the independence from `castAsRealFunctionClass.getFunction()`, and consistence with `builtinCastStringAsRealSig.
Moreover, I added an unit test with binary literal input,

BTW, The benchmark with `args=builtinCastRealAsRealSig` also showed that the new unit test with binary literal input went into `builtinCastRealAsRealSig.evalReal()`.
```
> go test -v -benchmem -bench=BenchmarkVectorizedBuiltinCastFunc -run=BenchmarkVectorizedBuiltinCastFunc -args builtinCastRealAsRealSig
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinCastFunc/builtinCastRealAsRealSig-VecBuiltinFunc-12         	12208030	        93.4 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCastFunc/builtinCastRealAsRealSig-NonVecBuiltinFunc-12      	   68865	     17330 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCastFunc/builtinCastRealAsRealSig-VecBuiltinFunc#01-12      	 2797251	       428 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCastFunc/builtinCastRealAsRealSig-NonVecBuiltinFunc#01-12   	   39289	     32114 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/pingcap/tidb/expression	5.861s
```